scripts/buildorder.py

```diff
@@ -347,19 +329,63 @@ def generate_target_buildorder(target_path, pkgs_map, fast_build_mode):
         package.deps.difference_update([subpkg.name for subpkg in package.subpkgs])
     return package.recursive_dependencies(pkgs_map)
 
+def get_list_cyclic_dependencies(pkgs_map, index=None, checked=None, pkgname=None, return_list=False):
+    "Find and print (or return a list of) circular dependencies for all packages or for one specified package."
+    # is_root - indicates that the function was run for the first time (called root function)
+    is_root = index == None
+    if is_root:
+        checked = []
+        index = []
+    result = []
+
+    # Start search
+    if is_root:
+        if pkgname:
+            range_pkgs = [pkgname]
+        else:
+            range_pkgs = pkgs_map.keys()
+    else:
+        range_pkgs = pkgs_map[index[-1]].deps
+    for pkg in range_pkgs:
+        if pkg in checked:
+            continue
+        index.append(pkg)
+        if index.count(pkg) == 2:
+            result.append(index.copy() if pkgname else index[index.index(pkg)::])
+        else:
+            result += get_list_cyclic_dependencies(pkgs_map, index, checked, pkgname)
+        del index[-1]
+        checked.append(pkg)
+
+    # return result
+    if is_root and not return_list:
+        # If launched in a root function and the `return_list` variable is set to False,
+        # then the result will be printed
+        if len(result) == 0:
+            print("No cyclic dependencies were found")
+        else:
+            print(f"Found {len(result)} cyclic dependencies:")
+            for cycle in result:
+                print("- "+" -> ".join(cycle))
+        sys.exit(0)
```

Comment:
Did this `sys.exit()` the last time I saw it? This is just fundamentally unsound function design - what's the type of this function? Do you still want it if I insist on changing the name to `get_list_cyclic_dependencies_and_maybe_die` so that each callsite makes it clear that the program might suddenly stop running when it's called? (Don't do that.)

Literally no way I will sign off on this with this function in it, sorry. Make this function take arguments and return a result; print on that result and/or `sys.exit(0)` in the places you want to do that. Feel free to add a helper function for the printing.